### ISSUE TITLE
[FEAT]: Add VIP Expiration to TorrentLeech

### ISSUE THREAD

**Issue by Lockszmith-GH** [Prowlarr/Prowlarr#419]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

TorrentLeech is a private tracker with VIP subscription option. Currently prowlarr doesn't have a VIP Expiration date field for it.

### Describe the solution you'd like

Add VIP Expiration date field.

### Describe alternatives you've considered

The VIP Expiration date _does_ show up in Generic torznab

### Anything else?

nothing else.

[AB#1442](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1442)

**Comment by Lockszmith-GH** [comment]
I looked at the code a bit, and it seems `VipExpiration` is not a member of `IIndexerSettings`, but of `NewznabSettings`.
`TorznabSettings` is inheriting from `NewznabSettings` thus it gets the VIP Expiration field.

I'm thinking introducing VIP Expiration to any private torrent/nzb site will require an interface inherited from `IIndexerSettings`, maybe something like `IPrivateIndexerSettings` - but I might not understand the code structure enough yet.

I just stumbled on Prowlarr, and it is definitly a much better solution than anything else. A universal download manager, with support to the 'arr' world.

I doubt I'll have more time in the near future to dive into code, but I hope this is something that will be interesting enough for some contributor to resolve.

Keep up the absolutely awesome work
